pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -506,6 +532,16 @@ func (r *Reconciler) applyNewResources(ctx context.Context, log logr.Logger, ori
 			if err := injectLabels(obj.obj, labelsToInject); err != nil {
 				return fmt.Errorf("error injecting labels into object %q: %s", resource, err)
 			}
+			// ref https://kubernetes.io/docs/reference/labels-annotations-taints/#kubectl-k8s-io-restart-at
+			if err := preserveAnnotation("Deployment", "apps/v1", current, obj.obj, "kubectl.kubernetes.io/restartedAt", "spec", "template", "metadata", "annotations"); err != nil {
+				return fmt.Errorf("could not preserve annotation : %w", err)
+			}
+			if err := preserveAnnotation("DaemonSet", "apps/v1", current, obj.obj, "kubectl.kubernetes.io/restartedAt", "spec", "template", "metadata", "annotations"); err != nil {
+				return fmt.Errorf("could not preserve annotation : %w", err)
+			}
+			if err := preserveAnnotation("StatefulSet", "apps/v1", current, obj.obj, "kubectl.kubernetes.io/restartedAt", "spec", "template", "metadata", "annotations"); err != nil {
+				return fmt.Errorf("could not preserve annotation : %w", err)
+			}
```

Comment:
It seems conceptually wrong to add a merge functionality at this place and not in the `pkg/resourcemanager/controller/managedresource/merger.go` file.
Wouldn't it better to handle the merging of the annotation in the Pod template in the already existing merge funcs? 
For example, can't we add the logic for preserving the annotation in https://github.com/gardener/gardener/blob/6ba5765c27a1cb90e0fcb11764af7cfe6cc013cd/pkg/resourcemanager/controller/managedresource/merger.go#L148-L162?